test/controllers/HospitalStayBeginControllerSpec.scala

```diff
@@ -0,0 +1,130 @@
+
+class HospitalStayBeginControllerSpec extends SpecBase {
+
+  val whenDidHospitalStayBeginPage = injector.instanceOf[WhenDidHospitalStayBeginPage]
+
+  class Setup(authResult: Future[~[Option[AffinityGroup], Enrolments]]) {
+    reset(mockAuthConnector)
+    when(mockAuthConnector.authorise[~[Option[AffinityGroup], Enrolments]](
+      Matchers.any(), Matchers.any[Retrieval[~[Option[AffinityGroup], Enrolments]]]())(
+      Matchers.any(), Matchers.any())
+    ).thenReturn(authResult)
+
+    val controller: HospitalStayBeginController = new HospitalStayBeginController(
+      mainNavigator,
+      whenDidHospitalStayBeginPage
+    )(authPredicate, dataRequiredAction, appConfig, mcc)
+
+    when(mockDateTimeHelper.dateTimeNow).thenReturn(LocalDateTime.of(2020, 2, 1, 0, 0, 0))
+  }
+
+  "HospitalStayBeginController" should {
+    "onPageLoadForWhenDidHospitalStayBegin" when {
+
+      "the user is authorised" must {
+
+        "return OK and correct view" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(userRequestWithCorrectKeys)
+          status(result) shouldBe OK
+        }
+
+        "return OK and correct view (pre-populated date when present in session)" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(fakeRequestConverter(fakeRequestWithCorrectKeys
+            .withSession(SessionKeys.whenHospitalStayBegin -> "2021-01-01")))
+          status(result) shouldBe OK
+          val documentParsed = Jsoup.parse(contentAsString(result))
+          documentParsed.select(".govuk-date-input__input").get(0).attr("value") shouldBe "1"
+          documentParsed.select(".govuk-date-input__input").get(1).attr("value") shouldBe "1"
+          documentParsed.select(".govuk-date-input__input").get(2).attr("value") shouldBe "2021"
+        }
+
+        "user does not have the correct session keys" in new Setup(AuthTestModels.successfulAuthResult) {
+          val result: Future[Result] = controller.onPageLoadForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
+          status(result) shouldBe INTERNAL_SERVER_ERROR
+        }
+      }
+
+      "the user is unauthorised" when {
+        "return 403 (FORBIDDEN) when user has no enrolments" in new Setup(AuthTestModels.failedAuthResultNoEnrolments) {
+          val result: Future[Result] = controller.onSubmitForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
+          status(result) shouldBe FORBIDDEN
+        }
+
+        "return 303 (SEE_OTHER) when user can not be authorised" in new Setup(AuthTestModels.failedAuthResultUnauthorised) {
+          val result: Future[Result] = controller.onSubmitForWhenDidHospitalStayBegin(NormalMode)(fakeRequest)
+          status(result) shouldBe SEE_OTHER
+        }
+      }
+    }
+
+    "onSubmitForWhenDidHospitalStayBegin" should {
+
+      "the user is authorised" must {
```

Comment:
I think some tests are missing here. We will need some tests around 'user enters a valid date and the key is stored in the session - redirecting to has health issue ended page' and 'users enters an invalid date' etc.
Some examples here: https://github.com/hmrc/penalties-appeals-frontend/blob/9b24c50e7ac470a00627108e48a278fa5998b7c2/test/controllers/CrimeReasonControllerSpec.scala#L98-L186